metron-platform/metron-common/src/test/java/org/apache/metron/common/dsl/functions/StringFunctionsTest.java

```diff
@@ -32,124 +33,143 @@
-        }
-        Assert.assertTrue(thrown);
-        thrown = false;
-
-        // NON INTEGER LENGTH
-        try {
-            run("FILL_RIGHT('123','X', 'z' )", new HashedMap());
-        }catch(ParseException pe){
-            thrown = true;
-            Assert.assertTrue(pe.getMessage().contains("not a valid Integer"));
-        }
-        Assert.assertTrue(thrown);
-        thrown = false;
-
-        // EMPTY STRING PAD
-        try {
-            Object returnValue = run("FILL_RIGHT('123','', 10 )", new HashedMap());
-        }catch(ParseException pe) {
-            thrown = true;
-            Assert.assertTrue(pe.getMessage().contains("cannot be an empty"));
-        }
-        Assert.assertTrue(thrown);
-        thrown = false;
-
-        //MISSING LENGTH PARAMETER
-        try {
-            run("FILL_RIGHT('123',foo)", variableMap);
-        }catch(ParseException pe){
-            thrown = true;
-            Assert.assertTrue(pe.getMessage().contains("expects three"));
-        }
-        Assert.assertTrue(thrown);
+    Assert.assertTrue(thrown);
+    thrown = false;
+
+    // NULL FILL
+    try{
+      run("FILL_RIGHT('123',foo, 7)", variableMap);
+    }catch(ParseException pe) {
+      thrown = true;
+      Assert.assertTrue(pe.getMessage().contains("are both required"));
+    }
+    Assert.assertTrue(thrown);
+    thrown = false;
+
+    // NON INTEGER LENGTH
+    try {
+      run("FILL_RIGHT('123','X', 'z' )", new HashedMap());
+    }catch(ParseException pe){
+      thrown = true;
+      Assert.assertTrue(pe.getMessage().contains("not a valid Integer"));
+    }
+    Assert.assertTrue(thrown);
+    thrown = false;
+
+    // EMPTY STRING PAD
+    try {
+      Object returnValue = run("FILL_RIGHT('123','', 10 )", new HashedMap());
+    }catch(ParseException pe) {
+      thrown = true;
+      Assert.assertTrue(pe.getMessage().contains("cannot be an empty"));
+    }
+    Assert.assertTrue(thrown);
+    thrown = false;
+
+    //MISSING LENGTH PARAMETER
+    try {
+      run("FILL_RIGHT('123',foo)", variableMap);
+    }catch(ParseException pe){
+      thrown = true;
+      Assert.assertTrue(pe.getMessage().contains("expects three"));
     }
+    Assert.assertTrue(thrown);
+  }
+
+  @Test
+  public void shannonEntropyTest() throws Exception {
+    //test empty string
+    Assert.assertEquals(0.0, (Double)run("STRING_ENTROPY('')", new HashMap<>()), 1e-6);
+    Assert.assertEquals(0.0, (Double)run("STRING_ENTROPY(foo)", ImmutableMap.of("foo", "")), 1e-6);
+
+    /*
+    Now consider the string aaaaaaaaaabbbbbccccc or 10 a's followed by 5 b's and 5 c's.
+    The probabilities of each character is as follows:
+    p(a) = 1/2
+    p(b) = 1/4
+    p(c) = 1/4
+    so the shannon entropy should be
+      -p(a)*log_2(p(a)) - p(b)*log_2(p(b)) - p(c)*log_2(p(c)) =
+      -0.5*-1 - 0.25*-2 - 0.25*-2 = 1.5
+     */
+    Assert.assertEquals(1.5, (Double)run("STRING_ENTROPY(foo)", ImmutableMap.of("foo", "aaaaaaaaaabbbbbccccc")), 1e-1);
```

Comment:
Answering both questions around equality bounds here.  The short answer was that it was just a mistake.  I've changed both of them to `Double.MIN_VALUE`.  In those calculations, it should be as near to exact as a double precision number gets.  I would have dropped the bounds, but the non-epsilon'd `assertEquals` is deprecated.  I guess, which would you prefer, a deprecated call, an `assertTrue` or an assertEquals with the minimum possible epsilon?  I chose the final one, but if you think otherwise, I'm not married to that decision.